### ISSUE TITLE
shdancer -video bgfx ACCESS VIOLATION

### ISSUE THREAD

**Issue by oomek** [mamedev/mame#5973]
0.215 has some problems when launching `shdancer` with BGFX. All BGFX backends seem affected.

OpenGL
```
-----------------------------------------------------
Exception at EIP=0000000003a42f53 (chain_manager::update_screen_textures(unsigned int, render_primitive*, osd_window&)+0x0b13): ACCESS VIOLATION
While attempting to read memory at ffffffffffffffff
-----------------------------------------------------
RAX=00400fff00400003 RBX=0000000012f49d70 RCX=0000000026fd09b0 RDX=00000000082c7dd0
RSI=0000000026fb0450 RDI=0000000000000007 RBP=000000000fb485a0 RSP=000000000fb48520
 R8=0000000000000000  R9=0000000000000000 R10=000000000000000f R11=000000000fb48690
R12=000000000fb48630 R13=0000000026fd09b0 R14=000000000fb48660 R15=0000000000000000
-----------------------------------------------------
Stack crawl:
  000000000fb48680: 0000000003a42f53 (chain_manager::update_screen_textures(unsigned int, render_primitive*, osd_window&)+0x0b13)
  000000000fb48820: 0000000003a1eba3 (renderer_bgfx::draw(int)+0x0123)
  000000000fb48880: 00000000039df845 (win_window_info::draw_video_contents(HDC__*, bool)+0x0055)
  000000000fb48950: 00000000039e2e92 (win_window_info::video_window_proc(HWND__*, unsigned int, unsigned long long, long long)+0x0732)
  000000000fb48980: 00000000039e41fd (winwindow_video_window_proc_ui(HWND__*, unsigned int, unsigned long long, long long)+0x000d)
  000000000fb48b10: 00007ffb802063ed (CallWindowProcW+0x03bd)
  000000000fb48b60: 00007ffb802060be (CallWindowProcW+0x008e)
  000000000fb48bd0: 00007ffb6f033919 (glPushClientAttrib+0x101a9)
  000000000fb48d60: 00007ffb802063ed (CallWindowProcW+0x03bd)
  000000000fb48dc0: 00007ffb80205fbc (DispatchMessageW+0x03cc)
  000000000fb48e20: 00007ffb80212753 (IsWindowVisible+0x01a3)
  000000000fb48ea8: 00007ffb808ffe14 (KiUserCallbackDispatcher+0x0024)
  000000000fb48eb0: 00007ffb7dbe1184 (NtUserMessageCall+0x0014)
  000000000fb48f50: 00007ffb80205a59 (SendMessageW+0x03e9)
  000000000fb48fb0: 00007ffb80205768 (SendMessageW+0x00f8)
  000000000fb49000: 00000000039dfaa1 (win_window_info::update()+0x01f1)
  000000000fb49050: 00000000039dec20 (windows_osd_interface::update(bool)+0x0050)
  000000000fb490d0: 0000000005639738 (video_manager::frame_update(bool)+0x00c8)
  000000000fb49130: 00000000056050d7 (screen_device::vblank_begin()+0x01b7)
  000000000fb491a0: 0000000005608d35 (screen_device::device_timer(emu_timer&, unsigned int, int, void*)+0x0175)
  000000000fb49250: 0000000005600c2a (device_scheduler::timeslice()+0x017a)
  000000000fb49390: 00000000055bd938 (running_machine::run(bool)+0x0248)
  000000000fb4f070: 0000000003a8b7e8 (mame_machine_manager::execute()+0x01d8)
  000000000fb4f300: 0000000003afc331 (cli_frontend::start_execution(mame_machine_manager*, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&)+0x01a1)
  000000000fb4f600: 0000000003afc62b (cli_frontend::execute(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >&)+0x005b)
  000000000fb4f660: 0000000003a898ba (emulator_info::start_frontend(emu_options&, osd_interface&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >&)+0x002a)
  000000000fb4fe20: 0000000009b38d1d (main+0x017d)
  000000000fb4fef0: 00000000004013a5 (__tmainCRTStartup+0x0225)
  000000000fb4ff20: 000000000040150b (mainCRTStartup+0x001b)
  000000000fb4ff50: 00007ffb7ff57bd4 (BaseThreadInitThunk+0x0014)
  000000000fb4ffd0: 00007ffb808cced1 (RtlUserThreadStart+0x0021)`

```
D3D11
```
-----------------------------------------------------
Exception at EIP=00000a7961720003 (register_frame_ctor+0x57a2cf93): ACCESS VIOLATION
While attempting to write memory at 00000a7961720003
-----------------------------------------------------
RAX=00000a7961720003 RBX=000000000fc56a90 RCX=0000000020c90810 RDX=00000000082c7dd0
RSI=000000002451a5a0 RDI=0000000000000007 RBP=000000000fb48940 RSP=000000000fb488b8
 R8=0000000000000000  R9=0000000000000000 R10=000000000000000f R11=000000000fb48a30
R12=000000000fb489d0 R13=0000000020c90810 R14=000000000fb48a00 R15=0000000000000000
-----------------------------------------------------
Stack crawl:
  000000000fb488b0: 00000a7961720003 (register_frame_ctor+0x57a2cf93)
  000000000fb48aa0: 0000000003a42f55 (chain_manager::update_screen_textures(unsigned int, render_primitive*, osd_window&)+0x0b15)
  000000000fb48bc0: 0000000003a1eba3 (renderer_bgfx::draw(int)+0x0123)
  000000000fb48c20: 00000000039df845 (win_window_info::draw_video_contents(HDC__*, bool)+0x0055)
  000000000fb48cf0: 00000000039e2e92 (win_window_info::video_window_proc(HWND__*, unsigned int, unsigned long long, long long)+0x0732)
  000000000fb48d20: 00000000039e41fd (winwindow_video_window_proc_ui(HWND__*, unsigned int, unsigned long long, long long)+0x000d)
  000000000fb48eb0: 00007ffb802063ed (CallWindowProcW+0x03bd)
  000000000fb48f50: 00007ffb802059fc (SendMessageW+0x038c)
  000000000fb48fb0: 00007ffb80205768 (SendMessageW+0x00f8)
  000000000fb49000: 00000000039dfaa1 (win_window_info::update()+0x01f1)
  000000000fb49050: 00000000039dec20 (windows_osd_interface::update(bool)+0x0050)
  000000000fb490d0: 0000000005639738 (video_manager::frame_update(bool)+0x00c8)
  000000000fb49130: 00000000056050d7 (screen_device::vblank_begin()+0x01b7)
  000000000fb491a0: 0000000005608d35 (screen_device::device_timer(emu_timer&, unsigned int, int, void*)+0x0175)
  000000000fb49250: 0000000005600c2a (device_scheduler::timeslice()+0x017a)
  000000000fb49390: 00000000055bd938 (running_machine::run(bool)+0x0248)
  000000000fb4f070: 0000000003a8b7e8 (mame_machine_manager::execute()+0x01d8)
  000000000fb4f300: 0000000003afc331 (cli_frontend::start_execution(mame_machine_manager*, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&)+0x01a1)
  000000000fb4f600: 0000000003afc62b (cli_frontend::execute(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >&)+0x005b)
  000000000fb4f660: 0000000003a898ba (emulator_info::start_frontend(emu_options&, osd_interface&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >&)+0x002a)
  000000000fb4fe20: 0000000009b38d1d (main+0x017d)
  000000000fb4fef0: 00000000004013a5 (__tmainCRTStartup+0x0225)
  000000000fb4ff20: 000000000040150b (mainCRTStartup+0x001b)
  000000000fb4ff50: 00007ffb7ff57bd4 (BaseThreadInitThunk+0x0014)
  000000000fb4ffd0: 00007ffb808cced1 (RtlUserThreadStart+0x0021)
````

verbose
```
D:\EMU\MAMEDEV>mame64 shdancer -v
Attempting load of mame.ini
Parsing mame.ini
Attempting load of mame.ini
Parsing mame.ini
Attempting load of horizont.ini
Attempting load of arcade.ini
Attempting load of raster.ini
Parsing raster.ini
Attempting load of source\segas18.ini
Attempting load of shdancer.ini
Starting plugin data...
Attempting load of mame.ini
Parsing mame.ini
Attempting load of mame.ini
Parsing mame.ini
Attempting load of horizont.ini
Attempting load of arcade.ini
Attempting load of raster.ini
Parsing raster.ini
Attempting load of source\segas18.ini
Attempting load of shdancer.ini
Video: Monitor                                                            65537 = "\\.\DISPLAY1" (primary)
Physical width 1920, height 1080
RawInput: APIs detected
RawInput: APIs detected
RawInput: APIs detected
DirectSound: Primary buffer: 48000 Hz, 16 bits, 2 channels
Input: Adding keyboard #0: HID Keyboard Device (device id: \\?\HID#VID_1532&PID_0226&MI_00#7&615d162&0&0000#{884b96c3-56ef-11d1-bc8c-00a0c91405dd})
Input: Adding keyboard #1: Razer Huntsman Elite (device id: \\?\HID#VID_1532&PID_0226&MI_01&Col01#7&1945eda0&0&0000#{884b96c3-56ef-11d1-bc8c-00a0c91405dd})
Input: Adding keyboard #2: HID Keyboard Device (device id: \\?\HID#VID_1532&PID_0060&MI_02#7&16d80d4a&0&0000#{884b96c3-56ef-11d1-bc8c-00a0c91405dd})
Input: Adding keyboard #3: Razer Lancehead Tournament Edition (device id: \\?\HID#VID_1532&PID_0060&MI_01&Col01#7&3a7f10c&0&0000#{884b96c3-56ef-11d1-bc8c-00a0c91405dd})
Input: Adding mouse #0: Razer Lancehead Tournament Edition (device id: \\?\HID#VID_1532&PID_0060&MI_00#7&f882b32&0&0000#{378de44c-56ef-11d1-bc8c-00a0c91405dd})
Input: Adding mouse #1: Razer Huntsman Elite (device id: \\?\HID#VID_1532&PID_0226&MI_02#7&2c7609de&1&0000#{378de44c-56ef-11d1-bc8c-00a0c91405dd})
Input: Adding lightgun #0: Razer Lancehead Tournament Edition (device id: \\?\HID#VID_1532&PID_0060&MI_00#7&f882b32&0&0000#{378de44c-56ef-11d1-bc8c-00a0c91405dd})
Input: Adding lightgun #1: Razer Huntsman Elite (device id: \\?\HID#VID_1532&PID_0226&MI_02#7&2c7609de&1&0000#{378de44c-56ef-11d1-bc8c-00a0c91405dd})
Region ':maincpu' created
unzip: opened archive file roms\shdancer.zip
unzip: found roms\shdancer.zip ECD
unzip: roms\shdancer.zip has no ZIP64 ECD locator
unzip: read roms\shdancer.zip central directory
unzip: closing archive file roms\shdancer.zip and sending to cache
unzip: found roms\shdancer.zip in cache
unzip: opened archive file roms\shdancer.zip
unzip: closing archive file roms\shdancer.zip and sending to cache
Region ':gfx1' created
unzip: found roms\shdancer.zip in cache
unzip: opened archive file roms\shdancer.zip
unzip: closing archive file roms\shdancer.zip and sending to cache
unzip: found roms\shdancer.zip in cache
unzip: opened archive file roms\shdancer.zip
unzip: closing archive file roms\shdancer.zip and sending to cache
unzip: found roms\shdancer.zip in cache
unzip: opened archive file roms\shdancer.zip
unzip: closing archive file roms\shdancer.zip and sending to cache
Region ':sprites' created
unzip: found roms\shdancer.zip in cache
unzip: opened archive file roms\shdancer.zip
unzip: closing archive file roms\shdancer.zip and sending to cache
unzip: found roms\shdancer.zip in cache
unzip: opened archive file roms\shdancer.zip
unzip: closing archive file roms\shdancer.zip and sending to cache
unzip: found roms\shdancer.zip in cache
unzip: opened archive file roms\shdancer.zip
unzip: closing archive file roms\shdancer.zip and sending to cache
unzip: found roms\shdancer.zip in cache
unzip: opened archive file roms\shdancer.zip
unzip: closing archive file roms\shdancer.zip and sending to cache
unzip: found roms\shdancer.zip in cache
unzip: opened archive file roms\shdancer.zip
unzip: closing archive file roms\shdancer.zip and sending to cache
unzip: found roms\shdancer.zip in cache
unzip: opened archive file roms\shdancer.zip
unzip: closing archive file roms\shdancer.zip and sending to cache
unzip: found roms\shdancer.zip in cache
unzip: opened archive file roms\shdancer.zip
unzip: closing archive file roms\shdancer.zip and sending to cache
unzip: found roms\shdancer.zip in cache
unzip: opened archive file roms\shdancer.zip
unzip: closing archive file roms\shdancer.zip and sending to cache
Region ':soundcpu' created
unzip: found roms\shdancer.zip in cache
unzip: opened archive file roms\shdancer.zip
unzip: closing archive file roms\shdancer.zip and sending to cache
unzip: found roms\shdancer.zip in cache
unzip: opened archive file roms\shdancer.zip
unzip: closing archive file roms\shdancer.zip and sending to cache
Optional device ':upd3' not found
Optional device ':upd2' not found
Optional device ':upd1' not found
Optional device ':mcu' not found
Optional memory region ':nvram' not found
Optional memory region ':screen' not found
Optional shared pointer ':rotateram' not found
Starting Shadow Dancer (World) ':'
  (missing dependencies; rescheduling)
Starting Motorola MC68000 ':maincpu'
Starting Zilog Z80 ':soundcpu'
Starting NVRAM ':nvram'
Starting Sega 315-5195 Memory Mapper ':mapper'
Starting Sega 315-5296 I/O ':io'
Starting Sega 315-5313 Megadrive VDP ':gen_vdp'
  (missing dependencies; rescheduling)
Starting palette ':gen_vdp:palette_lut'
Starting Sega VDP PSG ':gen_vdp:snsnd'
Starting palette ':gen_vdp:gfx_palette'
Starting palette ':gen_vdp:gfx_palette_shadow'
Starting palette ':gen_vdp:gfx_palette_hilight'
Starting Timer ':scantimer'
Starting Video Screen ':screen'
  (missing dependencies; rescheduling)
Starting gfxdecode ':gfxdecode'
Starting palette ':palette'
Starting Sega System 16B Sprites ':sprites'
Starting Sega 16-bit Video ':segaic16vid'
  (missing dependencies; rescheduling)
Starting Speaker ':mono'
  (missing dependencies; rescheduling)
Starting YM3438 OPN2C ':ym1'
Starting YM3438 OPN2C ':ym2'
Starting Ricoh RF5C68 ':rfsnd'
Starting Shadow Dancer (World) ':'
  (missing dependencies; rescheduling)
Starting Sega 315-5313 Megadrive VDP ':gen_vdp'
  (missing dependencies; rescheduling)
Starting Video Screen ':screen'
Starting Sega 16-bit Video ':segaic16vid'
Starting Speaker ':mono'
  (missing dependencies; rescheduling)
Starting Shadow Dancer (World) ':'
  (missing dependencies; rescheduling)
Starting Sega 315-5313 Megadrive VDP ':gen_vdp'
Starting Speaker ':mono'
Starting Shadow Dancer (World) ':'
Attempting to parse: default.cfg
Attempting to parse: shdancer.cfg
```

**Comment by cuavas** [comment]
I'd guess this is a duplicate of #5857 - have you tried a build made after that was fixed?

**Comment by oomek** [comment]
From what I see 0.215 is the latest and it crashes. I've tried 0.214 and it runs ok

**Comment by rb6502** [comment]
Yes, there have been many fixes made to the bgfx backend since 0.215.

**Comment by angelosa** [comment]
Cannot reproduce, closing.
